packages/flutter/lib/src/material/raised_button.dart

```diff
@@ -31,6 +31,45 @@ import 'theme_data.dart';
 /// Raised buttons have a minimum size of 88.0 by 36.0 which can be overidden
 /// with [ButtonTheme].
 ///
+/// {@tool sample}
+///
+/// This sample shows how to render a disabled RaisedButton, an enabled RaisedButton
+/// and lastly a RaisedButton with gradient background.
+///
+/// ```dart
+/// Column(
```

Comment:
I tried this example on an Android phone. The disabled button disappears behind the phone's status bar. Also: the sample generator expects a build method.

This version uses a Scaffold/Center/Column like the first example. Fixed up the formatting a little, and used Container's padding property instead of creating a Padding widget.

```
/// Widget build(BuildContext context) {
///   return Scaffold(
///     body: Center(
///       child: Column(
///         mainAxisSize: MainAxisSize.min,
///         children: <Widget>[
///           RaisedButton(
///             onPressed: null,
///             child: const Text('Disabled Button'),
///           ),
///           RaisedButton(
///             onPressed: () {},
///             child: const Text('Enabled Button'),
///           ),
///           RaisedButton(
///             onPressed: () {},
///             textColor: Colors.white,
///             padding: const EdgeInsets.all(0.0),
///             child: Container(
///               decoration: const BoxDecoration(
///                 gradient: LinearGradient(
///                   colors: <Color>[Colors.red, Colors.green, Colors.blue],
///                 ),
///               ),
///               padding: const EdgeInsets.all(10.0),
///               child: Text('Gradient Button'),
///             ),
///           ),
///         ],
///       ),
///     ),
///   );
/// }
```

Reply:
Out of town right now. Will update the code once back. Thanks

Reply:
@HansMuller 

> the sample generator expects a build method.

So every sample needs needs a build method? I saw some samples ( Container ), in which sample do not contain that method. 

I'll update the positioning of the widget. Also, can you please review the IconButton (https://github.com/flutter/flutter/pull/27169/commits/3af1b401c84bc935fd74f18b99f03f9b8f374024) sample as well.

Reply:
You're right that sample code marked with `{@tool sample}` doesn't need a build method.  In this case, since the sample code can stand alone as an app, it should be marked with `{@tool snippet --template=stateless_widget}`, so that complete app gets generated (too).

The IconButton sample looks good.